testgen/generators/collation_short.py

```diff
@@ -152,136 +283,110 @@ def generateCollTestData2(self, filename, icu_version, start_count=0):
                 locale = locale_match.group(1)
                 rules = []
                 locale = ""
-                attributes = []
-                strength = None
+                attributes = {}
+                case_first = None
+                case_level = None
+                numeric = None
+                backwards = None
+
+                line_number += 1
+
                 continue
 
-            # Find "** test" section
-            is_test = test_line.match(line_in)
-            if is_test:
-                test_description = is_test.group(1)
+            # Find "** test" section. Simply reset the description but leave rules alone.
+            is_test_line = self.test_line.match(line_in)
+            if is_test_line:
+                # Get the description for subsequent tests
+                test_description = is_test_line.group(1)
+                line_number += 1
                 continue
 
             # Handle rules, to be applied in subsequent tests
-            is_rules = rule_header_pattern.match(line_in)
+            is_rules = self.rule_header_pattern.match(line_in)
             if is_rules:
```

Comment:
Can you add a line explaining that seeing a new rule string means creating a new collator instance?